CMakeLists.txt

```diff
@@ -463,7 +463,7 @@ endif ()
 
 # libwasm, which implenents the wasm C API
 if (BUILD_LIBWASM)
-  add_library(wasm SHARED ${WABT_LIBRARY_SRC} src/interp/interp-wasm-c-api.cc)
+  add_library(wasm ${WABT_LIBRARY_SRC} src/interp/interp-wasm-c-api.cc)
   target_link_libraries(wasm PUBLIC wabt)
   target_include_directories(wasm PUBLIC third_party/wasm-c-api/include)
   if (COMPILER_IS_MSVC)
```

Comment:
I'm guessing the problem with removing `SHARED` is with these definitions... you don't want `__declspec(dllexport)` on static library code; those macros should just be empty.

Reply:
`WASM_API_EXTERN` should only be set when the library is being built at shared, yes.  So we should had a check here if we want to make it flexible.

Still.. I don't understand why this library isn't defaulting to `SHARED` if that is the cmake default.

Reply:
I may have misunderstood the cmake default in my initial statement. It may as well be, that the default is to build statically, in which case this would would indeed cause a change.